typescript/web/src/components/export-button/export-modal/utils/get-dataset-name.ts

```diff
@@ -0,0 +1,18 @@
+import { ExportFormat } from "../../../../graphql-types/globalTypes";
+
+export const getDatasetExportName = ({
+  datasetSlug,
+  format,
+}: {
+  datasetSlug: string;
+  format: ExportFormat;
+}) => {
+  const dateObject = new Date();
+  // toLocaleDateString("fr") format is "01/11/1993"
+  const [day, month, year] = dateObject.toLocaleDateString("fr").split("/");
+  const date = `${year}-${month}-${day}`;
+  // toLocaleTimeString("fr") format is "14:31:32"
+  const time = dateObject.toLocaleTimeString("fr").replace(/:/g, "");
+  const dateAndTime = `${date}T${time}`;
```

Comment:
Since this code uses the `fr` locale, if the end-user does not lives in France, the time (and possibly the date) would be set to an incorrect value.
As the moment lib is already needed by some of our dependencies, you can `pushd typescript/web && yarn add moment && popd` to add the package directly then use `moment().format('YYYY-MM-DDTHH:MM:SS')` instead of the whole block.

Reply:
> Since this code uses the `fr` locale, if the end-user does not lives in France, the time (and possibly the date) would be set to an incorrect value.

Are you sure about that ? My understanding of the argument `locales` is precisely to force the format to be 'fr' compliant, regardless of where you are located.

> As the moment lib is already needed by some of our dependencies, you can pushd typescript/web && yarn add moment && popd to add the package directly then use moment().format('YYYY-MM-DDThh:mm:ss') instead of the whole block.

I would strongly advocate against that lib, [as the creators themselves do](https://momentjs.com/docs/#/-project-status/), especially for such a simple usage.

Reply:
> Are you sure about that ?

I was. But I've read the function documentation too fast and I was wrong:

> The toLocaleDateString() method returns a string with a language sensitive representation of the date portion of the specified date in the user agent's timezone.

But still, relying on this function then parsing it is not optimal compared to the previous version.

> > As the moment lib is already needed by some of our dependencies, you can pushd typescript/web && yarn add moment && popd to add the package directly then use moment().format('YYYY-MM-DDThh:mm:ss') instead of the whole block.

>
> I would strongly advocate against that lib, [as the creators themselves do](https://momentjs.com/docs/#/-project-status/), especially for such a simple usage. If we really need a lib, we can use [date-fns](https://date-fns.org/), seems better suited

OK I see... Since it's already used by the common-resolvers and release package, let's use date-fns if you don't find a vanilla alternative to the `moment.format` function 👍

Reply:
done